omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java

```diff
@@ -110,11 +114,24 @@ public String get(PageModel model,
 			Context.removeProxyPrivilege(GET_LOCATIONS);
 		}
 
+		model.addAttribute("showSessionLocations", !isLocationUserPropertyAvailable(administrationService));
+		Object showLocation = pageRequest.getAttribute("showSessionLocations");
+		if(showLocation != null && showLocation.toString().equals("true")) {
```

Comment:
Are you sure we are expecting this value to be true?

Reply:
Yes, Initial login URL will be "http://localhost:8080/openmrs/login.htm", and then if the user doesn't have the location userproperty, then it will redirect to "http://localhost:8080/openmrs/referenceapplication/login.page?showSessionLocations=true&"